airflow/plugins/operators/sentry_to_gcs.py

```diff
@@ -0,0 +1,200 @@
+        }
+
+        for future in as_completed(futures):
+            combined_response_data.extend(future.result())
+
+    combined_response_data = [
+        x
+        for x in combined_response_data
+        if pendulum.from_format(
+            x["dateCreated"][:-1], "YYYY-MM-DDTHH:mm:ss"
+        ).to_date_string()
+        == target_date
+    ]
+    return combined_response_data
+
+
+def fetch_and_clean_from_sentry(extract, target_date, auth_token):
+    """
+    Download Sentry event records as a DataFrame.
+    """
+
+    print(f"Downloading Sentry event data for project {extract.project_slug}")
+    all_rows = iterate_over_sentry_records(extract, target_date, auth_token)
+    extract.extract_ts = pendulum.now()
+    extract.data = all_rows
+
+    return extract
+
+
+class SentryExtract(PartitionedGCSArtifact):
+    bucket: ClassVar[str] = CALITP_BUCKET__SENTRY_EVENTS
+    table: ClassVar[str] = "events"
+    dt: pendulum.Date
+    ts: pendulum.DateTime
+    partition_names: ClassVar[List[str]] = ["dt", "ts"]
+    project_slug: str
+    data: Optional[List[Dict]]
+    extract_ts: Optional[pendulum.DateTime]
+
+    def save_to_gcs(self, fs):
+        raw_df = pd.DataFrame([{**make_arrays_bq_safe(row)} for row in self.data])
+        cleaned_df = raw_df.rename(make_name_bq_safe, axis="columns")
+
+        self.save_content(
+            fs=fs,
+            content=gzip.compress(
+                cleaned_df.to_json(orient="records", lines=True).encode()
+            ),
+            exclude={"data"},
+        )
+
+
+class SentryToGCSOperator(BaseOperator):
+    template_fields = ("bucket",)
+
+    def __init__(
+        self,
+        bucket,
+        project_slug,
+        auth_token=None,
+        target_date=None,
+        **kwargs,
+    ):
+        """
+        An operator that downloads data from a specific issue within a Sentry instance
+        and saves it as a JSON file hive-partitioned by date and time in Google Cloud
+        Storage (GCS).
+
+        Args:
+            bucket (str): GCS bucket where the scraped Sentry issue will be saved.
+            project_slug (str): The identifier for the Sentry project being examined.
+            auth_token (str, optional): The auth token to use when downloading from Sentry.
+                This can be someone's personal auth token. If not provided, the environment
+                variable of `CALITP_SENTRY_AUTH_TOKEN` is used.
+        """
+        self.bucket = bucket
+        self.project_slug = project_slug
+        self.auth_token = auth_token
+        self.target_date = target_date
+
+        super().__init__(**kwargs)
+
+    def execute(self, **kwargs):
+        ts = pendulum.now()
+        dt = ts.date()
+        extract = SentryExtract(
+            project_slug=self.project_slug,
+            dt=dt,
+            ts=ts,
+            filename=f"{self.project_slug}.jsonl.gz",
+        )
+        target_date = self.target_date or ts.subtract(days=1).to_date_string()
```

Comment:
It seems like `target_date` should really be `execution_date`, I think? Each task execution should fetch the same set of data, to facilitate backfills as well as catchups from the `start_date`.